docs/markdown/Writing Plugins/common-plugin-tasks/plugin-upgrade-guide.md

```diff
@@ -19,11 +19,11 @@ is now required in order to:
 
 Some deprecations are in place to reduce the burden on `@rule` authors:
 
-#### `Goal.environment_migrated`
+#### `Goal.environment_behavior`
```

Comment:
I think this might benefit from a more holistic rewrite. The second paragraph is misleading if you are fine with `LOCAL_ONLY`, which I suspect many plugin goals will be. It probably would make more sense to first briefly explain what environments are and what the deprecation is, then explain the simple case of LOCAL_ONLY, and finally get in how to migrate to the new EnvironmentName mechanism

Reply:
@Eric-Arellano See re-written docs. Any precision-related feedback would be appreciated, but I think the general form is much better.